CodeAcademyChat/CodeAcademyChat/Classes/Chat/Room.swift

```diff
@@ -9,22 +9,42 @@ import Foundation
 
 class Room {
     let name: String
-//    var onlineUsers: [User]
-//    var offlineUsers: [User]
-//    var messageHistory: [Message]
-    var message: [Message]
+    //    var onlineUsers: [User]
+    //    var offlineUsers: [User]
+    //    var messageHistory: [Message]
+    var messages: [Message]
     
     init(name: String,
-//         onlineUsers: [User],
-//         offlineUsers: [User],
-//         messageHistory: [Message],
-         message: [Message]) {
+         //         onlineUsers: [User],
+         //         offlineUsers: [User],
+         //         messageHistory: [Message],
+         messages: [Message]) {
         self.name = name
-//        self.onlineUsers = onlineUsers
-//        self.offlineUsers = offlineUsers
-//        self.messageHistory = messageHistory
-        self.message = message
+        //        self.onlineUsers = onlineUsers
+        //        self.offlineUsers = offlineUsers
+        //        self.messageHistory = messageHistory
+        self.messages = messages
     }
+    
+    func writeMessage(messageContent: String, sender: User) {
+        var message = Message(datetime: .now, username: sender.username, content: messageContent)
+        messages.append(message)
+    }
+    
+    func getMessages() -> [String] {
+        
+        guard  messages.count > 0 else {
+            return []
+        }
+        var messageList: [String] = []
+        
+        for message in messages {
+            let messageContent = message.content
+            messageList.append(messageContent)
+        }
+        return messageList
```

Comment:
Pasidomėkit kaip veikia .map funkcija, visai įdomus dalykas

Reply:
su map Tadas spreads. as pabandziau for issisukti